### DISCUSSION TITLE
Broadcast rate limit sleep events

### DISCUSSION THREAD

**Discussion by dmyers** [spatie/guzzle-rate-limiter-middleware#13] [Ideas]
It would be neat to be able to easily log and identify what is being rate limited and for how long if there was a way to get the current `store` and `deferrer` instances as event data.

**Comment by sebastiandedeyne** [comment]
I'd accept a PR that uses Guzzle's event system for this.

https://docs.guzzlephp.org/en/5.3/events.html
